utils/versions.py

```diff
@@ -0,0 +1,156 @@
+    )
+    return branch_version
+
+def create_vernacular_version(
+        collection_id: Union[int, str],
+        suffix: Optional[str] = None
+    ) -> str:
+    version_path = f"{collection_id}/"
+    return create_version(version_path, 'vernacular_metadata', suffix)
+
+def create_mapped_version(
+        vernacular_version: str, suffix: Optional[str] = None) -> str:
+    return create_version(vernacular_version, 'mapped_metadata', suffix)
+
+def create_validation_version(
+        mapped_version: str,
+        suffix: Optional[str] = None
+):
+    validation_version = create_version(mapped_version, 'validation', suffix)
+    return validation_version.rstrip('/') + ".csv"
+
+def create_content_data_version(
+        mapped_version: str, suffix: Optional[str] = None) -> str:
+    return create_version(mapped_version, 'content_data', suffix)
+
+def get_most_recent_vernacular_version(collection_id: Union[int, str]):
+    data_root = os.environ.get("VERNACULAR_DATA", "file:///tmp")
+    versions = storage.list_dirs(f"{data_root.rstrip('/')}/{collection_id}/")
+    if not versions:
+        raise Exception(
+            "No vernacular metadata versions found for {collection_id}")
+    recent_version = sorted(versions)[-1]
+    return f"{collection_id}/{recent_version}/"
+
+def get_pages(version: str, pipeline_step: str, **kwargs):
+    if pipeline_step == 'vernacular': 
+        data_root = os.environ.get("VERNACULAR_DATA", "file:///tmp")
+    elif pipeline_step == 'mapped':
+        data_root = os.environ.get("MAPPED_DATA", "file:///tmp")
+    else:
+        raise Exception("Invalid pipeline step")
+
+    data_path = f"{data_root.rstrip('/')}/{version.rstrip('/')}/data/"
+    try:
+        page_list = storage.list_pages(data_path, recursive=True, **kwargs)
+    except FileNotFoundError as e:
+        print(
+            f"\n\nNo {pipeline_step} pages found in {e.filename}\n\n"
+        )
+        raise(e)
+    return [path[len(data_root)+1:] for path in page_list]
+
+def get_vernacular_pages(version, **kwargs):
+    return get_pages(version, 'vernacular_metadata', **kwargs)
+
+def get_mapped_pages(version, **kwargs):
+    return get_pages(version, 'mapped_metadata', **kwargs)
+
+def get_child_directories(version, **kwargs):
+    data_root = os.environ.get('MAPPED_DATA', "file:///tmp")
+    child_directories = storage.list_dirs(
+        f"{data_root.rstrip('/')}/{version.rstrip('/')}/data/",
+        recursive=False
+    )
+    return child_directories
+
+def get_child_pages(version, **kwargs):
+    data_root = os.environ.get("MAPPED_DATA", "file:///tmp")
+    data_path = f"{data_root.rstrip('/')}/{version.rstrip('/')}/data/children/"
+    try:
+        page_list = storage.list_pages(data_path, recursive=False, **kwargs)
+    except FileNotFoundError:
+        return []
+    except OSError:
+        return []
+    return [path[len(data_root)+1:] for path in page_list]
+
+def get_vernacular_page(version_page):
```

Comment:
I think `get_vernacular_page_content` would be clearer as a name, and distinguish this function's purpose from `get_vernacular_pages`. 

You could also rename `get_vernacular_pages` --> `get_vernacular_page_list`? I think it's ok to leave as is too.